### DISCUSSION TITLE
Convergences issues with Newton for coupled density-dependent groundwater flow-transport model

### DISCUSSION THREAD

**Discussion by robinkeegan** [MODFLOW-ORG/modflow6#2019] [General]
I have a coupled groundwater flow and transport model with density. The model simulates an unconfined aquifer using 8 layers. During the simulation some of the upper layers become unsaturated. As the cells dry and later re-wet, I thought it would be best to enable Newton options. 
The model runs fine with Newton enabled when there is no density. However, when I enable density the model has convergence issues after a few timesteps. Looking at the ims .csv the problematic cells (solution_outer_dvmax_node) are in layers above the watertable (unsaturated). The model (with density) runs fine when Newton is switched off. 
The easy option would be to use the inbuilt rewetting and disable Newton but I am curious is there a reason Newton doesn't play nice with density and any workarounds? I have tried enabling with_relaxation but that didn't make a difference.  
I also tested the MODFLOW example "ex-gwt-henry-a" with Newton enabled and it does not converge. 
I am using MODFLOW 6.5.0
Thanks,
Robin

**Comment by christianlangevin** [comment]
Hi @robinkeegan, I've run and tested the density capabilities in MODFLOW 6 with the Newton formulation.  While there can be some additional challenges with convergence, I haven't run into too many problems.  If you can post the Newton version of ex-gwt-henry-a that is giving you trouble, I can take a look.  There is a simplified Newton example [here](https://github.com/langevin-usgs/modflow-training-princeton2024/blob/main/examples/notebooks_completed/variable-density-tidal.ipynb) that shows a working MODFLOW 6 model with the Buoyancy Package for a rather challenging problem with cell wetting and drying.

**Comment by vivekbedekar** [comment]
@robinkeegan - Adding NEWTON indeed caused convergence issues in this example, however, the simulation worked when I added NO_PTC to flow.ims.
